### ISSUE TITLE
changing mount point from /mnt/c to /c

### ISSUE THREAD

**Issue by raymod2** [Microsoft/BashOnWindows#1918]
Years of using MSYS and MSYS2 has ingrained the habit of referring to the C: drive as "/c" but Bash on Windows uses "/mnt/c".  A good solution is using a bind mount:

`mkdir -p /c`
`sudo mount -o bind /mnt/c /c`

However, this requires root and needs to be run every time you open a bash window.  I tried using /etc/fstab for a persistent mount but I couldn't get it to work:

`sudo sh -c "echo '/mnt/c   /c   none   bind' >> /etc/fstab`

**Comment by jack775544** [comment]
As a workaround you could possibly edit your sudoers file to allow any user to use access `mount` without root privileges and then add that command to your profile script.

**Comment by Vaporeon** [comment]
Why not just `ln -s /mnt/c /c`?
I personally think /mnt/*x* is a more sane default.

**Comment by raymod2** [comment]
Yeah a soft link works, too.  One minor gripe with a soft link is that "ls /c" doesn't work.  You need "ls /c/".

**Comment by raymod2** [comment]
BTW, this ticket is to look into why /etc/fstab doesn't work.

**Comment by aseering** [comment]
I imagine this is because WSL doesn't perform most normal Linux systemwide initialization.  See #511 , for example.

**Comment by sgtoj** [comment]
@raymod2 Here the solution...

```
sudo mkdir -p /c 
sudo sh -c "echo '/mnt/c /c none bind' >> /etc/fstab"
sudo mount -a
```

**Comment by therealkenc** [comment]
It's the `sudo mount -a` that people don't want to run, per what Adam alludes right above. `fstab` is not respected on WSL instantiation.

It is tempting to dupe these all into #511 ("make `systemd` work") but now there is precedent in #2530 for adding `systemd` features to WSL `init`. I hope that isn't the route this goes, but we'll have to let it play out.

**Comment by sgtoj** [comment]
@therealkenc You're correct. I should have posted my answered after testing closing and relaunching a session. Once I noticed I was wrong, I started putting something together to as work around to share with my team and others.

**Comment by sgtoj** [comment]
My original solution is the _proper_ linux solution. However, the solution currently does not work for Ubuntu for Windows because mounts are not auto loaded. See @therealkenc comments for more information.

In the meantime,  I have documented and posted a short [step-by-step guide](https://gist.github.com/sgtoj/f82990bcd9e89db49b84e2d2e70b281d) to implement a workaround.

**Comment by therealkenc** [comment]
Ref #2636

**Comment by d1egoaz** [comment]
FYI this is now possible to do it with `wsl.conf` https://docs.microsoft.com/en-us/windows/wsl/wsl-config
`root = /`

```
diego@diego-thinkpad:~$ cat /etc/wsl.conf
# Enable extra metadata options by default
[automount]
enabled = true
root = /
options = "metadata,umask=22,fmask=11"
mountFsTab = false

# Enable DNS – even though these are turned on by default, we’ll specify here just to be explicit.
[network]
generateHosts = true
generateResolvConf = true

diego@diego-thinkpad:~$ mount -l | grep drvfs
C: on /c type drvfs (rw,noatime,uid=1000,gid=1000,umask=22,fmask=11,metadata)
diego@diego-thinkpad:~$
```
![image](https://user-images.githubusercontent.com/950087/36636323-73cbd262-1979-11e8-95d3-01a8c612dd83.png)

**Comment by Nossiac** [comment]
@d1egoaz How to upgrade the latest OS build? The update center gave me " Version 1709, OS build 16299", seems not the one your had mentioned.

**Comment by lemmy04** [comment]
...where is that wsl.conf file supposed to be...?

**Comment by jonaskuske** [comment]
> ...where is that wsl.conf file supposed to be...?

`/etc/wsl.conf` :)
